scala/README.md

```diff
@@ -534,22 +620,73 @@ Tips & Tricks
 
     ```scala
     // Value classes
-    case class Email(email: String) extends AnyVal // no extra object allocation at runtime
+    case class Email(email: String) extends AnyVal // No extra object allocation at runtime
 
-    val email = Email("no@one.com") // never gets mixed with regular strings
+    val email = Email("joe@example.com") // Never gets mixed with other strings
 
     // Extension methods
     implicit class IntOps(val n: Int) extends AnyVal {
       def stars = "*" * n
     }
 
-    5.stars // equivalent to a static method call, no implicit conversion actually takes place
+    5.stars // Equivalent to a static method call, no implicit conversion actually takes place
     ```
 
 0. Leverage parallel collections, use `.par` judiciously.
 
 0. Use the `@tailrec` annotation to ensure the compiler can recognize a recursive method is tail-recursive.
 
+0. You can use the `@scala.beans.BeanProperty` and `@BooleanBeanProperty` annotations to automatically generate JavaBeans style getter and setter methods.
+
+0. Use this syntax to pass a sequence as parameter to a variable length argument list method:
+
+    ```scala
+    def foo(args: Int*) = ???
+
+    val seq = Seq(1, 2, 3)
+
+    foo(seq:_*)
+    ```
+
+0. There are some really neat tricks we can do with pattern matching:
+
+    ```scala
+    val tuple = ("foo", 1, false)
+    val (x, y, z) = tuple
+    assert((x, y, z) == (("foo", 1, false)))
```

Comment:
@matthandlersux it does with `-Yno-adapted-args`, which all our projects must use:

```scala
$ scala
Welcome to Scala version 2.11.7 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_51).

scala> assert((1, 2) == (1, 2))

$ scala -Yno-adapted-args

scala> assert((1, 2) == (1, 2))
<console>:11: error: too many arguments for method ==: (x$1: Any)Boolean
       assert((1, 2) == (1, 2))
                     ^

scala> assert((1, 2) == ((1, 2)))

```

It is a minor annoyance, but with a big payoff. Many puzzlers just vanish:

```scala
$ scala

scala> Seq(1,2,3).toSet()
res0: Boolean = false

scala> Set.empty()
res1: Boolean = false

$ scala -Yno-adapted-args

scala> Seq(1,2,3).toSet()
<console>:11: error: not enough arguments for method apply: (elem: AnyVal)Boolean in trait GenSetLike.
Unspecified value parameter elem.
       Seq(1,2,3).toSet()
                       ^

scala> Set.empty()
<console>:11: error: not enough arguments for method apply: (elem: Unit)Boolean in trait GenSetLike.
Unspecified value parameter elem.
       Set.empty()
                ^
```

@a-wortman can beat you in the head till are you convinced it is good :-)